src/util/tmpl/fd_treap.c

```diff
@@ -0,0 +1,946 @@
+/* Generate prototypes, inlines and implementations for ultra high
+   performance treaps.  A treap hybrid of a binary search tree and heap
+   such that it will well-balanced on average statistically.
+
+   It is not as well-balanced theoretically as more complex
+   non-randomized balancing tree algorithms (e.g. red-black trees, AVL
+   trees, etc).  But it is often better in practice as it is much
+   simpler (i.e. amenable for ultra high performance and small code
+   footprint implementation) and very adaptable (e.g. easy to tweak to
+   support adaptive queries like a splay tree, etc).  Additionally,
+   there are a bunch of tricks in the below to optimize this much
+   further than textbook implementations (those tend to miss a lot of
+   practical opportunities including eliminating the cost of eliminating
```

Comment:
```suggestion
   practical opportunities including eliminating the cost of 
```
I can barely read the code but this seems like a typo?

Reply:
Whoops ... missed this before but will be checking in some stuff shortly and will fix the typo then.

Reply:
Fixed in https://github.com/firedancer-io/firedancer/pull/353